overlord/devicestate/devicemgr.go

```diff
@@ -78,6 +78,19 @@ var (
 // during managers' startup.
 var EarlyConfig func(st *state.State, preloadGadget func() (sysconfig.Device, *gadget.Info, error)) error
 
+// ErrNoDeviceIdentityYet is returned when the device doesn't have a serial assertion.
+// It's a special case of ErrNoState.
+type ErrNoDeviceIdentityYet struct{}
```

Comment:
nitpick: probably want to make the struct *Error unexported and use a singleton for the error

Reply:
Thanks. I've fixed this.